### DISCUSSION TITLE
How do I cancel other operations in Async.Parallel when one operation returns

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1304] [General]
I am trying to make http requests that may return something or post something
Let's say

```F#
    type HttpOperation =
    | Get of Async<string>
    | Put of Async<unit>
```

I have a bunch of ```Get``` or ```Put``` in ```seq<HttpOperation>```, and I wanted to execute them all together for efficiency via ```Async.Parallel```

```
async {
    let! results = (mySeq :Async<string> seq) |> Async.Parallel
}
```

but instead of getting results, I wanted the first result available and cancel other operations in Async.Parallel, now how do I do the cancelation?

**Comment by Tarmil** [comment]
Sounds like you want `Async.Choice`, rather than `Async.Parallel`.

(Also this kind of question would be more suited for StackOverflow than fslang-suggestions)

**Reply by Xyncgas** [reply]
I tried searching stack overflow and chatGPT before I asked a human

**Reply by bartelink** [reply]
The point is that the other humans don't ask these questions in a low traffic language design _suggestions_ (that have been well thought out) forum
They ask them on a Q&A site about solving specific problems when using a thing: Stack overflow. Seriously, there are way more watchers on that than this forum
Imagine if everyone asked all F# usage questions on this language design discussions forum!

**Reply by Happypig375** [reply]
There are also the F# Slack and the F# Discord if you want to write to people directly.
